homeassistant/components/camera/push.py

```diff
@@ -75,8 +82,18 @@ def __init__(self, hass, image_field):
 
         if _camera is None:
             _LOGGER.error("Unknown %s", entity_id)
+            status = HTTP_NOT_FOUND if request[KEY_AUTHENTICATED]\
+                else HTTP_UNAUTHORIZED
             return self.json_message('Unknown {}'.format(entity_id),
-                                     HTTP_BAD_REQUEST)
+                                     status)
+
+        authenticated = (request[KEY_AUTHENTICATED] or
+                         request.query.get('token') == _camera.token)
```

Comment:
Not setting a token will result in token being `None`, so then not passing a token means we're checking `None == None`, allowing user to always access it.